client/blocks/login/index.jsx

```diff
@@ -46,6 +47,12 @@ class Login extends Component {
 		}
 	};
 
+	componentWillReceiveProps = ( nextProps ) => {
+		if ( this.props.twoFactorAuthType !== nextProps.twoFactorAuthType ) {
```

Comment:
@gziolo: I'm a bit behind discussions about notices. Looking at the `renderError()` method it seems this component is able to handle both global and local notices. Now I have no idea if it's a left over or if that's something that is required. Maybe @scruffian or @drewblaisdell can chime in.